### ISSUE TITLE
Possibility to wait until condition passed

### ISSUE THREAD

**Issue by sebi3110** [truong-bui/AsyncLoadingScreen#103]
Hi there,

great work with the plugin!

I have one question:
Is it possible to wait for another condition (apart from `PostLoadMapWithWorld`) before the loading screen disappears?
I want to have the loading screen on until `FShaderPipelineCache::NumPrecompilesRemaining()` is equal to 0

Would be awesome if this would be possible.

Best regards :)

**Comment by truong-bui** [comment]
Hi there,

Unfortunately, it is not possible without modifying the Engine source because the Async Loading Screen plugin depends on the MoviePlayer module, which binds to the `PostLoadMapWithWorld `delegate.

**Comment by mBerdis** [comment]
Hi, it would be much appreciate to implement this. It's needed since 5.3 because of PSO precaching. Without waiting object pop up as their shaders get compiled and it looks bad.
https://dev.epicgames.com/documentation/unreal-engine/pso-precaching-for-unreal-engine#loading-screen

**Comment by truong-bui** [comment]
@mBerdis, yes, I know about the PSO. I plan to implement this feature in the next version. But I can't guarantee it will be done soon, as I am a bit busy right now.

**Comment by truong-bui** [comment]
@sebi3110 @mBerdis @SkyKingPX Async Loading Screen v1.7.0 now fully support PSO Precaching loading. Please update your plugin to latest version
<img width="1920" height="1080" alt="image" src="https://github.com/user-attachments/assets/15c96587-c508-4a42-b185-d0d129379b61" />
- Changlogs: https://github.com/truong-bui/AsyncLoadingScreen/tree/master#changelog
- Tutorial: https://github.com/truong-bui/AsyncLoadingScreen#enable-pso-precaching-progress-widget

**Comment by SkyKingPX** [comment]
Amazing, thanks!
